packages/expo-modules-core/ios/JSI/JavaScriptObject.mm

```diff
@@ -0,0 +1,71 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+#import <ExpoModulesCore/JSIConversions.h>
+#import <ExpoModulesCore/JavaScriptObject.h>
+#import <ExpoModulesCore/JavaScriptRuntime.h>
+#import <ExpoModulesCore/ExpoModulesProxySpec.h>
+
+@implementation JavaScriptObject {
+  __weak JavaScriptRuntime *_runtime;
+  std::shared_ptr<jsi::Object> _jsObjectPtr;
+}
+
+- (nonnull instancetype)initWith:(std::shared_ptr<jsi::Object>)jsObject
+                         runtime:(nonnull JavaScriptRuntime *)runtime
+{
+  if (self = [super init]) {
+    _runtime = runtime;
+    _jsObjectPtr = jsObject;
+  }
+  return self;
+}
+
+- (jsi::Object *)get
+{
+  return _jsObjectPtr.get();
+}
+
+#pragma mark - Subscripting
+
+- (nullable id)objectForKeyedSubscript:(nonnull NSString *)key
+{
+  auto runtime = [_runtime get];
+  auto callInvoker = [_runtime callInvoker];
+
+  if (runtime && callInvoker) {
+    auto value = _jsObjectPtr->getProperty(*runtime, [key UTF8String]);
+    return expo::convertJSIValueToObjCObject(*runtime, value, callInvoker);
+  }
+  return nil;
+}
+
+- (void)setObject:(nullable id)obj forKeyedSubscript:(nonnull NSString *)key
+{
+  if (auto runtime = [_runtime get]) {
```

Comment:
Wojtek says it's terrible 😅

```suggestion
  auto runtime = [_runtime get];

  if (runtime) {
```

Reply:
I would even go with `early return` pattern here 👀 
```suggestion
  auto runtime = [_runtime get];
  
  if (!runtime) {
    return;
  }
```